### PR TITLE
fix(amplify-category-function): adds policy for list Bucket for user groups

### PR DESCRIPTION
**Pull request by akshbhu**
*Issue #3509 

*Description of changes:*
* added policy specific to Cognito user groups for List Bucket operation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ async function addWalkthrough(context, defaultValuesFilename, serviceMetadata, o
   while (!checkIfAuthExists(context)) {
     if (
       await context.amplify.confirmPrompt.run(
-        'You need to add auth (Amazon Cognito) to your project in order to add storage for user files. Do you want to add auth now?'
+        'You need to add auth (Amazon Cognito) to your project in order to add storage for user files. Do you want to add auth now?',
```
Comment (kaustavghosh06) [inline packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js]: Why do we need the extra ,?

File: packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js [hunk 2]
```diff
@@ -168,7 +168,7 @@ async function configure(context, defaultValuesFilename, serviceMetadata, resour
       if (permissionSelected === 'Learn more') {
         context.print.info('');
         context.print.info(
-          'You can restrict access using CRUD policies for Authenticated Users, Guest Users, or on individual Groups that users belong to in a User Pool. If a user logs into your application and is not a member of any group they will use policy set for “Authenticated Users”, however if they belong to a group they will only get the policy associated with that specific group.'
+          'You can restrict access using CRUD policies for Authenticated Users, Guest Users, or on individual Groups that users belong to in a User Pool. If a user logs into your application and is not a member of any group they will use policy set for “Authenticated Users”, however if they belong to a group they will only get the policy associated with that specific group.',
```
Comment (kaustavghosh06) [inline packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js]: Same as above.

File: packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js [hunk 3]
```diff
@@ -544,6 +544,28 @@ async function updateCfnTemplateWithGroups(context, oldGroupList, newGroupList,
       },
     };
   });
+  // added a new policy for the user group to make action on buckets
+  newGroupList.forEach(group => {
```
Comment (kaustavghosh06) [inline packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js]: I don't think this is the right way to go about this. Which "CRUD" operation is this IAM policy tied to? In this implementation, it seems we are adding it to all of the "CRUD" operations.

Comment (akshbhu) [reply packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js]: In the above we are adding permissions for objects not on buckets when we do bucketname/*. 
In this to use S3::listbuckets operation , we need IAM policy on bucket.

Comment (kaustavghosh06) [reply packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js]: We should associate this policy only if the the selection has "Read" as a part of the CRUD flow.

Comment (akshbhu) [reply packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js]: Do you mean this line
```javascript
NewGroupPolicyMap[group].includes('s3:ListBucket') === true
```
Here I am checking if any action is there on bucket , I am adding the policy then.

Comment (kaustavghosh06) [reply packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js]: Ah! got it, somehow I missed it.

**Pull request merged**